sdk/storage/azure-storage-stress/src/main/java/com/azure/storage/stress/TelemetryHelper.java

```diff
@@ -0,0 +1,306 @@
+import io.opentelemetry.instrumentation.runtimemetrics.java8.MemoryPools;
+import io.opentelemetry.instrumentation.runtimemetrics.java8.Threads;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder;
+import io.opentelemetry.sdk.trace.data.LinkData;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import io.opentelemetry.sdk.trace.samplers.SamplingResult;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+
+/**
+ * Telemetry helper is used to monitor test execution and record stats.
+ */
+public class TelemetryHelper {
+    private final Tracer tracer;
+    private final ClientLogger logger;
+    private static final AttributeKey<String> SCENARIO_NAME_ATTRIBUTE = AttributeKey.stringKey("scenario_name");
+    private static final AttributeKey<String> ERROR_TYPE_ATTRIBUTE = AttributeKey.stringKey("error.type");
+    private static final AttributeKey<Boolean> SAMPLE_IN_ATTRIBUTE = AttributeKey.booleanKey("sample.in");
+    private static final OpenTelemetry OTEL;
+    private final String scenarioName;
+    private final Meter meter;
+    private final DoubleHistogram runDuration;
+    private final Attributes commonAttributes;
+    private final Attributes canceledAttributes;
+    private final String packageType;
+    private final String packageVersion;
+
+    private final AtomicLong successfulRuns = new AtomicLong();
+    private final AtomicLong failedRuns = new AtomicLong();
+
+    static {
+        // enables micrometer metrics from Reactor schedulers allowing to monitor thread pool usage and starvation
+        Schedulers.enableMetrics();
+        OTEL = init();
+    }
+
+    /**
+     * Creates an instance of telemetry helper.
+     * @param scenarioClass the scenario class
+     */
+    public TelemetryHelper(Class<?> scenarioClass) {
+        this.scenarioName = scenarioClass.getName();
+        this.tracer = OTEL.getTracer(scenarioName);
+        this.meter = OTEL.getMeter(scenarioName);
+        this.logger = new ClientLogger(scenarioName);
+        this.runDuration = meter.histogramBuilder("test.run.duration")
+            .setUnit("s")
+            .build();
+        this.commonAttributes = Attributes.of(SCENARIO_NAME_ATTRIBUTE, scenarioName);
+        this.canceledAttributes = Attributes.of(SCENARIO_NAME_ATTRIBUTE, scenarioName, ERROR_TYPE_ATTRIBUTE, "cancelled");
+        this.packageType = scenarioClass.getPackage().toString();
+        // Since packageVersion is not obtainable, hardcoding the values based on the package type.
+        if (packageType.contains("blob")) {
+            this.packageVersion = "12.25.1";
```

Comment:
I don't think hardcoding is helpful - it's really easy to forget to update it and would lead to more confusion at the results.

If we can't find a way to get a real package version, leaving it empty or removing the attribute would be a better choice